### PR TITLE
Feature/backend leaderboard

### PR DESCRIPTION
**Pull request by luis-prates**
Changes DB gameStats json field to separate fields. Improves GET games/active so it allows the requester to get both Waiting and In Progress games or just one of them. Also protects against empty values. Adds tests for some game HTTP endpoints. Adds GET leaderboard endpoint and tests.

The leaderboard ranks users first by their won games, then by their lost games. If two users have the same number of won games, the user with the least number of lost games will be ranked higher. If two users have the same number of won and lost games, their ranks will be the same

### CODE REVIEW

**Comment by luis-prates** [comment]
Needs to be synced with dev after other pull request #21 is approved and merged unless this one is to be reviewed first, then it's the other way around. As draft until then.

**Review by Hevhove — APPROVED** [review]
Looks good. Thanks for fixing my tests.

**Pull request merged**
